langchain4j-pinecone/src/main/java/dev/langchain4j/store/embedding/pinecone/PineconeEmbeddingStore.java

```diff
@@ -2,73 +2,83 @@
-     *                    The ID can be found in the Pinecone URL: https://app.pinecone.io/organizations/.../projects/...:{projectId}/indexes.
-     * @param index       The name of the index (e.g., "test").
-     * @param nameSpace   (Optional) Namespace. If not provided, "default" will be used.
-     * @param metadataTextKey   (Optional) The key to find the text in the metadata. If not provided, "text_segment" will be used.
+     * @param apiKey          The Pinecone API key.
+     * @param cloud           The cloud (e.g., "AWS").
+     * @param region          The region (e.g., "us-east-1").
+     * @param index           The name of the index (e.g., "test").
+     * @param nameSpace       (Optional) Namespace. If not provided, "default" will be used.
+     * @param metadataTextKey (Optional) The key to find the text in the metadata. If not provided, "text_segment" will be used.
+     * @param metadataTypeMap (Optional) The remaining metadata type map to store metadata. If not provided, will not store metadata.
+     * @param dimension       (Optional) The dimension of embedding aims to create index. If not provided, will not create index
+     * @param createIndex     (Optional) Whether to create index or not. If not provided "false" will be used
      */
     public PineconeEmbeddingStore(String apiKey,
-                                  String environment,
-                                  String projectId,
+                                  String cloud,
+                                  String region,
                                   String index,
                                   String nameSpace,
-                                  String metadataTextKey) {
-
-        PineconeClientConfig configuration = new PineconeClientConfig()
-                .withApiKey(apiKey)
-                .withEnvironment(environment)
-                .withProjectName(projectId);
-
-        PineconeClient pineconeClient = new PineconeClient(configuration);
-
-        PineconeConnectionConfig connectionConfig = new PineconeConnectionConfig()
-                .withIndexName(index);
-
-        this.connection = pineconeClient.connect(connectionConfig);
+                                  String metadataTextKey,
+                                  Map<String, Value.KindCase> metadataTypeMap,
+                                  Integer dimension,
+                                  Boolean createIndex) {
+        Pinecone client = new Pinecone.Builder(apiKey).build();
         this.nameSpace = nameSpace == null ? DEFAULT_NAMESPACE : nameSpace;
         this.metadataTextKey = metadataTextKey == null ? DEFAULT_METADATA_TEXT_KEY : metadataTextKey;
+        this.metadataTypeMap = metadataTypeMap;
+
+        // create serverless index if not exist
+        if (Boolean.TRUE.equals(createIndex) && index != null && !isIndexExist(client, index)) {
+            client.createServerlessIndex(index, "cosine", dimension, cloud, region);
```

Comment:
worth checking that `index`, `dimension`, `cloud` and `region` are provided (e.g. `ensureNotBlank`, `ensureNotNull`, etc) if `createIndex==true`

Reply:
done.